test/test_formats.py

```diff
@@ -158,6 +157,16 @@ def test_ratings(self):
             loaded_metadata = save_and_load_metadata(self.filename, metadata)
             self.assertEqual(int(loaded_metadata['~rating']), rating, '~rating: %r != %r' % (loaded_metadata['~rating'], rating))
 
+    def test_guess_format(self):
+        if self.original:
+            fd, temp_file = mkstemp()
+            os.close(fd)
+            shutil.copy(self.original, temp_file)
+            audio = picard.formats.guess_format(temp_file)
+            audio_original = picard.formats.open(self.filename)
+            self.assertEqual(type(audio), type(audio_original))
+            os.unlink(temp_file)
```

Comment:
There's a risk of remaining temp files in case of failures, registring a cleanup method is much safer, see https://docs.python.org/2/library/unittest.html#unittest.TestCase.addCleanup
It can be done in another PR though.